app/models/cherrypick_task.rb

```diff
@@ -153,18 +219,40 @@ def perform_pick(requests, robot)
     source_plates = Set.new
     plates_hash = build_plate_wells_from_requests(requests) # array formed from requests
 
+    # Initial settings needed for control requests addition
+    if control_plate
+      num_plate = 0
+      batch = requests.first.batch
+      control_assets = control_plate.wells.joins(:samples)
+      control_positions = control_positions(batch.id, num_plate, current_destination_plate.size, control_assets.count)
+    end
+
     push_completed_plate = lambda do
       destination_plates << current_destination_plate.completed_view
       current_destination_plate = yield # reset to start picking to a fresh one
+      if control_plate
```

Comment:
```suggestion
      if auto_add_control_plate
```

We both got a bit confused with the name of this variable, cos it sounds like it's saying that the current source plate is a control plate. Can we change it to something like the above? Thanks

Reply:
Renaming with a better variable name